assets/dev/js/editor/components/icons-manager/modal-content.js

```diff
@@ -0,0 +1,353 @@
+export default class extends Marionette.ItemView {
+	id() {
+		return 'elementor-icons-manager';
+	}
+
+	cache() {
+		return {
+			value: '',
+			type: '',
+			icons: false,
+		};
+	}
+
+	getTemplate() {
+		return '#tmpl-elementor-icons-manager';
+	}
+
+	getDefaultSettings() {
+		return {
+			name: '',
+			url: '',
+			prefix: '',
+			displayPrefix: '',
+			ver: '',
+			enqueue: [],
+			exclude: [],
+			include: [],
+			icons: [],
+			fetchJson: false,
+		};
+	}
+
+	iconTypes() {
+		return [];
+	}
+
+	ui() {
+		return {
+			tabLabel: '.icon-type-tab-label',
+			searchInput: '#icons-search',
+			iconValueInput: '#icon_value',
+			iconTypeInput: '#icon_type',
+			tabLi: '.icon-type-tab-label',
+			iconListContainer: '.elementor-icon-manager-tabs-content',
+			iconListItems: '.icon-list-item',
+			iconListItem: '.icon-list-item i',
+		};
+	}
+
+	events() {
+		return {
+			'click @ui.tabLabel': 'onTabClick',
+			'click @ui.iconListItems': 'onIconClick',
+			'click @ui.iconListItem': 'onIconClick',
+			'input @ui.searchInput': 'onSearch',
+		};
+	}
+
+	isInCache( name ) {
+		return ( this.cache[ name ] );
+	}
+
+	toggleSearchBarVisibility( state ) {
+		return state ? this.ui.searchInput.show() : this.ui.searchInput.hide();
+	}
+
+	onTabClick( event ) {
+		this.showTab( jQuery( event.target ).attr( 'data-tab' ) );
+	}
+
+	showTab( tabKey ) {
+		const tabSettings = ElementorConfig.icons[ tabKey ];
+
+		tabSettings.name = tabKey;
+
+		if ( ! this.isInCache( tabKey ) ) {
+			this.initIconType( tabSettings );
+			this.cache[ tabKey ] = true;
+		}
+
+		this.ui.tabLi.removeClass( 'active' );
+
+		const icons = this.store().getIcons( tabSettings );
+		if ( ! icons ) {
+			return;
+		}
+		this.cache.type = tabKey;
+		jQuery( 'li[data-tab="' + tabKey + '"]' ).addClass( 'active' );
+		this.ui.iconListContainer.html( '' );
+		for ( const i in icons ) {
```

Comment:
1. You should build your items as a Marionette Collection View.
2. Never use plain `for` loop in JavaScript.